tools/hls-verifier/lib/HlsVhdlTb.cpp

```diff
@@ -6,664 +6,517 @@
-//     )
-//     port map(
-//       port1 => signal1,
-//       port2 => signal2
-//     );
-class Instance {
-  std::string moduleName;
-  std::string instanceName;
-  std::vector<std::pair<std::string, std::string>> connections;
-  std::vector<std::pair<std::string, std::string>> params;
-
-public:
-  Instance(std::string module, std::string inst,
-           const std::vector<unsigned> &p = {})
-      : moduleName(std::move(module)), instanceName(std::move(inst)) {}
-
-  Instance &parameter(const std::string &parameter, const std::string &value) {
-    params.emplace_back(parameter, value);
-    return *this;
-  }
-
-  Instance &connect(const std::string &port, const std::string &signal) {
-    connections.emplace_back(port, signal);
-    return *this;
-  }
-
-  void emitVhdl(mlir::raw_indented_ostream &os) {
-    os << instanceName << ": entity work." << moduleName << "\n";
-
-    // VHDL port map needs a continuous assignment
-    std::sort(connections.begin(), connections.end(),
-              [](const auto &a, const auto &b) { return a.first < b.first; });
-
-    if (!params.empty()) {
-      os << "generic map(\n";
-      os.indent();
-      for (size_t i = 0; i < params.size(); ++i) {
-        os << params[i].first << " => " << params[i].second;
-        if (i != params.size() - 1)
-          os << ",\n";
-      }
-      os.unindent();
-      os << ")\n";
+  void connectToDuv(Instance &duvInst) {
+    if (isInput) {
+      duvInst.connect(argName, argName + "_dout0")
+          .connect(argName + "_valid", argName + "_dout0_valid")
+          .connect(argName + "_ready", argName + "_dout0_ready");
+    } else {
+      duvInst.connect(argName, argName + "_din0")
+          .connect(argName + "_valid", argName + "_valid")
+          .connect(argName + "_ready", argName + "_ready");
     }
-    os << "port map(\n";
-    os.indent();
-    for (size_t i = 0; i < connections.size(); ++i) {
-      const auto &[port, sig] = connections[i];
-      os << port << " => " << sig;
-      if (i != connections.size() - 1)
-        os << ",";
-      os << "\n";
+  }
+};
+
+// A helper construct for interfacing the control only signals of the DUV
+//
+// HACK: the control only signals are handled in the following ways:
+// - Input: they are driven by valid = constant 1, their ready signals are
+// ignored
+// - Output: they are driving the join
+struct ControlOnlyConnector {
+
+  handshake::ControlType type;
+  std::string argName;
+  bool isInput;
+
+  ControlOnlyConnector(handshake::ControlType type, const std::string &argName,
+                       bool isInput)
+      : type(type), argName(argName), isInput(isInput) {}
+
+  void declareSignals(mlir::raw_indented_ostream &os) {
+    declareSTL(os, argName + "_valid");
+    declareSTL(os, argName + "_ready");
+  }
+
+  void connectToDuv(Instance &duvInst) {
+    if (isInput) {
+      duvInst.connect(argName + "_valid", "\'1\'")
+          .connect(argName + "_ready", argName + "_ready");
```

Comment:
```suggestion
          .connect(argName + "_ready", "open");
```

And you can remove signal declaration